app/code/core/Mage/Api/Model/Server/Adapter/Jsonrpc.php

```diff
@@ -83,10 +83,33 @@ public function run()
     {
         $this->_jsonRpc = new Zend_Json_Server();
         $this->_jsonRpc->setClass($this->getHandler());
+
+        // Allow soap_v2 style request.
+        $request = $this->_jsonRpc->getRequest();
+        $method = $request->getMethod();
+        if (!$this->_jsonRpc->getServiceMap()->getService($method)) {
+            // Convert request to v1 style.
+            $request->setMethod('call');
+            $params = $request->getParams();
+            $sessionId = $params[0] ?? null;
+            unset($params[0]);
+            $params = count($params)
+                ? [$sessionId, $method, $params]
+                : [$sessionId, $method];
+            $request->setParams($params);
+        }
+
         $this->getController()->getResponse()
             ->clearHeaders()
             ->setHeader('Content-Type', 'application/json; charset=utf8')
             ->setBody($this->_jsonRpc->handle());
+
+        Mage::dispatchEvent('api_server_adapter_jsonrpc_run_after', [
```

Comment:
sorry, I meant the reason why throwing that event

Reply:
It's for logging of incoming APIs. It's very useful during development, and settle disputes in production.

Reply:
other API types don't have something like that right?

Reply:
No, others don't have events.

Reply:
mmm as a thought, should we add such event to the other API? or, is it needed here if was never implemented elsewhere? sorry for bothering, trying to understand the rationale

Reply:
The event for other APIs should be added in separate PRs. Or if you prefer, it can be taken out here. This PR is what I have in my projects for over a year and it works for my purposes. If it is only useful for me and no others, then I do not have issue of not merging it. 

It does looks strange if we have the event for JSON-RPC but not SOAP/XML. On the other hand, it is a new API we are adding. New API, new event. From my tests, JSON-RPC is significantly faster than SOAP (900ms vs 1800ms). More reasons to migrate.

Reply:
let's go :-) I think we can do that later, in case somebody needs it :-)